### PR TITLE
Add private endpoint dns zone config

### PR DESCRIPTION
**Pull request by BlakeWills**
This PR closes #

The changes in this PR are as follows:

* Add private endpoint builder with "link_to_private_dns_zone" operators.
* Add depends_on support to subnet builder.

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [x] **Tested my code** end-to-end against a live Azure subscription.
* [x] **Updated the documentation** in the docs folder for the affected changes.
* [x] **Written unit tests** against the modified code that I have made.
* [x] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [x] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

Below is a minimal example configuration that includes the new features, which can be used to deploy to Azure:

- See the updated docs.

### CODE REVIEW

File: src/Farmer/Farmer.fsproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <!-- General -->
     <AssemblyName>Farmer</AssemblyName>
-    <Version>1.6.32</Version>
+    <Version>1.6.34.12-beta</Version>
```
Comment (BlakeWills) [inline src/Farmer/Farmer.fsproj:5]: Fix this

File: src/Farmer/Farmer.fsproj [hunk 2]
```diff
@@ -26,7 +26,7 @@
     <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
     <RepositoryType>git</RepositoryType>
     <RepositoryUrl>https://github.com/CompositionalIT/farmer</RepositoryUrl>
-
+	  <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
```
Comment (BlakeWills) [inline src/Farmer/Farmer.fsproj:29]: Remove this

File: src/Farmer/Arm/Network.fs [hunk 3]
```diff
@@ -18,6 +18,7 @@ let virtualNetworks = ResourceType ("Microsoft.Network/virtualNetworks", "2020-0
 let virtualNetworkGateways = ResourceType ("Microsoft.Network/virtualNetworkGateways", "2020-05-01")
 let localNetworkGateways = ResourceType ("Microsoft.Network/localNetworkGateways", "")
 let privateEndpoints = ResourceType ("Microsoft.Network/privateEndpoints", "2020-07-01")
+let privateEndpointsDnsZoneGroup = ResourceType ("Microsoft.Network/privateEndpoints/privateDnsZoneGroups", "2021-05-01")
```
Comment (TheRSP) [inline src/Farmer/Arm/Network.fs:21]: these are normally pluralised

File: src/Farmer/Builders/Builders.VirtualNetwork.fs [hunk 4]
```diff
@@ -409,6 +419,74 @@ type VNetPeeringSpecBuilder() =
...
+          PrivateLinkServiceConnection = None
+          PrivateDnsZone = None }
+    
+    [<CustomOperation "name">]
+    member _.Name(state:PrivateEndpointConfig, name:string) = { state with Name = (ResourceName name) }
+    member _.Name(state:PrivateEndpointConfig, name:ResourceName) = { state with Name = name }
+
+    [<CustomOperation "subnet">]
+    member _.Subnet(state:PrivateEndpointConfig, subnet:SubnetReference) = { state with Subnet = Some subnet }
+    member _.Subnet(state:PrivateEndpointConfig, subnet:SubnetConfig) = { state with Subnet = Some (SubnetReference.Direct (Managed (subnet :> IBuilder).ResourceId)) }
```
Comment (TheRSP) [inline src/Farmer/Builders/Builders.VirtualNetwork.fs:471]: If you moved this below the type extension on Lines 490-492, you could use `SubnetReference.create(subnet)` to simplify this expression

**Review by TheRSP — APPROVED** [review]

**Pull request merged**